ProductProvider.Infrastructure/Services/WarehouseService.cs

```diff
@@ -1,5 +1,332 @@
+            throw;
+        }
+    }
+
+    /// <summary>
+    /// Updates a unique warehouse product entry in the database.
+    /// This method modifies the properties of an existing warehouse product identified by the provided warehouseId.
+    /// 
+    /// It checks if the provided WarehouseModel contains new color and size information. If the color or size does not exist
+    /// in the database, it creates new entries for them. If they do exist, it updates the product with the corresponding
+    /// IDs. The method also updates the current stock of the product. 
+    /// 
+    /// Returns the updated WarehouseModel if successful, or null if the warehouse model is null, the warehouseId is empty, 
+    /// or the product was not found in the database.
+    /// 
+    /// <param name="newWarehouseModel">The new WarehouseModel containing the updated product details.</param>
+    /// <param name="warehouseId">The unique identifier for the warehouse product to be updated.</param>
+    /// <returns>The updated WarehouseModel or null.</returns>
+    public async Task<WarehouseModel?> UpdateUniqueProductAsync(WarehouseModel newWarehouseModel, Guid warehouseId)
+    {
+        if (newWarehouseModel == null || warehouseId == Guid.Empty)
+        {
+            Console.WriteLine("A valid WarehouseModel and a valid warehouseId must be provided.");
+            return null;
+        }
+        try 
+        {
+            await using var context = _contextFactory.CreateDbContext();
+            var warehouseEntity = await context.Warehouses.FirstOrDefaultAsync(w => w.UniqueProductId == warehouseId);
+
+            if(warehouseEntity == null)
+            {
+                Console.WriteLine($"The product with the id: {warehouseId} was not found");
+                return null;
+            }
+
+            if (warehouseEntity.ColorId == Guid.Empty) 
+            {
+                if (newWarehouseModel.Color == null)
+                {
+                    warehouseEntity.ColorId = newWarehouseModel.ColorId;
+                }
+                else if (newWarehouseModel.Color.Name != null || newWarehouseModel.Color.HexadecimalColor != null)
+                {
+                    var newColor = new ColorEntity
+                    {
+                        Name = newWarehouseModel.Color.Name,
+                        HexadecimalColor = newWarehouseModel.Color.HexadecimalColor
+                    };
+
+                    await context.Colors.AddAsync(newColor).ConfigureAwait(false);
+
+                    warehouseEntity.ColorId = newColor.Id;
+                }
+            }
+            else
+            {
+                warehouseEntity.ColorId = newWarehouseModel.ColorId;
+            }
+
+            if (warehouseEntity.SizeId == Guid.Empty)
+            {
+                if (newWarehouseModel.Size == null)
+                {
+                    warehouseEntity.SizeId = newWarehouseModel.ColorId;
```

Comment:
warehouseEntity.SizeId ska inte bli newWareHouseModel.ColorId?

Reply:
ny commit @Jonas-Dahlbo
[Corrected line 256 to warehouseEntity.SizeId = newWarehouseModel.SizeId;](https://github.com/Projekt-Grupp-2/ProductProvider/pull/29/commits/eef6b7ac9f1968cd1c8f72081234e27c6932cfe3)